runtime/src/main/scala/org/apache/pekko/grpc/internal/PekkoNettyGrpcClientGraphStage.scala

```diff
@@ -112,6 +112,7 @@ private final class PekkoNettyGrpcClientGraphStage[I, O](
         override def onMessage(message: O): Unit =
           callback.invoke(message)
         override def onClose(status: Status, trailers: Metadata): Unit = {
+          if (!matVal.isCompleted) onHeaders(trailers)
           trailerPromise.success(trailers)
```

Comment:
Hmm, this is a bit weird, putting the trailer headers *both* in the header *and* in the trailer field of the materialized value. I think we should probably do either - i.e. add an 'else'. 

That makes things a little more complicated for clients, who might have to look both at the headers and at the trailer headers to find e.g. the status. Perhaps we could create a helper function to make that easier at some point.

Reply:
Hmm, yeah I thought about that too. If we do that, sometimes `trailerPromise` is never completed. I am not super familiar with all the things that are happening here but I was afraid of leaking that. If we do that should we not also do something with `trailerPromise`? Either fail it or succeed it with empty metadata?

Reply:
I thought at least this way, I am being more honest because the trailers and the headers are the same thing

Reply:
`trailerPromise` never completing would indeed be bad, that's a great point!

Taking into account the spec at https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md, the case where everything is in the first HEADERS segments is called the 'Trailers-Only' case - so perhaps it would be most faithful complete `onHeaders` with empty metadata, and complete the `trailerPromise` with the received headers?

Reply:
Addressed in https://github.com/apache/pekko-grpc/pull/391/commits/b791d66abab4778d7cb178d4dc38e4dee3cf86e6